src/coq/decompiler/decompiler.ml

```diff
@@ -303,17 +356,57 @@ let rec first_pass env sigma (opts : (unit Proofview.tactic * string) list) trm
     (* "fun x => ..." -> "intro x." *)
     | Lambda (n, t, b) ->
        let (env', trm', names) = zoom_lambda_names env 0 trm in
+       Printf.printf "ADDED NAME TO CONTEXT\n";
        Compose ([ Intros names ], [ first_pass env' sigma opts trm' ])
     (* Match on well-known functions used in the proof. *)
     | App (f, args) ->
-       choose (rewrite <|> induction <|> left <|> right <|> split
+       choose (rewrite <|> induction <|> left <|> right(* <|> split*)
                <|> reflexivity <|> symmetry <|> exists) (f, args)
     (* Hypothesis transformations or generation tactics. *)
     | LetIn (n, valu, typ, body) ->
        choose (rewrite_in <|> apply_in <|> pose) (n, valu, typ, body)
     (* Remainder of body, simply apply it. *)
     | _ -> def
-         
+
+and try_custom_tacs env sigma all_opts trm =
```

Comment:
you probably also need to return a sigma here

Reply:
This is a little difficult because our "subterms_with_type" function returns a list with many sigmas. So we have a different sigma for each subgoal and then a corresponding call to first_pass. What can we do about this?

Reply:
OK let's just cut an issue for evar_map handling